### ISSUE TITLE
Can't configure custom storage

### ISSUE THREAD

**Issue by mkarnicki** [orchestral/testbench#194]
First off, thank you for creating this package. I'm developing my very first, and this has been extremely helpful. Maybe I'm doing something wrong here :).

## What I'm doing
My package involves sending mail. In `getEnvironmentSetUp` I configure mail.driver to use `log`.
Occasionally I would like to see the output of the e-mail, so I comment out `Mail::fake()`.

## What should happen
Having configured storage (as seen below), I would expect Testbench to put the logs within my designated `storage/logs` directory.

## What actually happens
Instead, the mails are still logged to the internal storage of the Testbench package at
`vendor/orchestra/testbench-core/laravel/storage/logs`

## My setup
```
    protected function getEnvironmentSetUp($app)
    {
        [...]
        // Setup mail
        $app['config']->set('mail.driver', 'log');
        // TODO: For some reason, this doesn't seem to work with Testbench.
        // Setup storage for mail testing to log files
        $app['config']->set('filesystems.disks.testing.driver', 'local');
        $app['config']->set('filesystems.disks.testing.root', realpath(__DIR__.'/../storage'));
        $app['config']->set('filesystems.default', 'testing');
    }
```

Hopefully I'm just doing something wrong :). Please advise, thank you.

**Comment by crynobone** [comment]
```php
$app['config']->set('filesystems.disks.testing.driver', 'local');
$app['config']->set('filesystems.disks.testing.root', realpath(__DIR__.'/../storage'));
$app['config']->set('filesystems.default', 'testing');
```

AFAIK `Mail::fake()` is using the default `Logger`, so I don't know what `filesystems.disks.testing` is doing, that is not part of anything Laravel is doing so therefore it should be handle by the userland.

**Comment by mkarnicki** [comment]
Hi @crynobone , thank you for your reply. Do note I mentioned 

> I comment out `Mail::fake()`

Hence, if you set the usual mail driver to `log`, sending the e-mail should just save it to the log file. And it does, just within orchestral/testbench directory, instead of my package's storage/log directory.

In the given example, 'testing' is just a disk. Instead, you could try something more default like this:

```
$app['config']->set('filesystems.default', 'local'); // the default Laravel setting, so this changes nothing
$app['config']->set('filesystems.disks.local.root', realpath(__DIR__.'/../storage'));
```

and the problem would still be reproducible. The above changes the storage directory (used for the logs subdirectory) that would otherwise be set by config/filesystems.php config file.

In other words, just like we're able to override the database config with
`$app['config']->set('database......`
we should be able to override where the logs are stored - but this doesn't currently work.

To me, the problem is valid. What do you think? Thanks.

**Comment by crynobone** [comment]
> To me, the problem is valid. What do you think? Thanks.

No, as I explained above this is not something that Testbench mess around with. What we do is simply bootstrap Laravel application for package tests.

**Comment by mkarnicki** [comment]
Okay, thanks man. I'll have a closer look at how `getEnvironmentSetUp` is used and see what's going on. Cheers.

**Comment by Artistan** [comment]
```php
$app->useStoragePath(realpath(__DIR__.'/../storage') );
```

**Comment by lucaspottersky** [comment]
Inspired by this thread, I ended up with this:

```php
protected function getEnvironmentSetUp($app) {
        $app->useStoragePath(realpath(__DIR__.'/../../storage/logs/laravel.log'));

        $app['config']->set('logging.channels.single', [
            'driver' => 'single',
            'path' => realpath(__DIR__.'/../../storage/logs/laravel.log'),
            'level' => 'debug',
        ]);
}
```

and now I can see the logs in my folder (instead of vendor). 
thanks.
